core/editor/data/favorites/collection.php

```diff
@@ -0,0 +1,62 @@
+<?php
+namespace Elementor\Core\Editor\Data\Favorites;
+
+use Elementor\Core\Utils\Collection as Base_Collection;
+
+abstract class Collection {
+
+	/**
+	 * The Laravel inspired Collection.
+	 *
+	 * @var Base_Collection|null
+	 */
+	protected $collection = null;
+
+	/**
+	 * Collection constructor.
+	 */
+	public function __construct() {
+		$this->collection = new Base_Collection( [] );
+	}
```

Comment:
> Why do we need collection inside the collection?

Actually I first thought that it's a good idea to extend the base Laravel-inspired collection, and then add the methods I need. But then I realised that most of the base-collection methods create and return a new collection instance. That's a problem, because at initialization process, when a new collection is registered, the Manager class keeps an instance of it so further actions can be taken. To register a collection again in the Manager every time a method called is logically wrong. So my collection handles this by creating a base-collection method and replacing it every time a method returns a new instance. That's by the way, why my collection should be dynamic.